### ISSUE TITLE
unusedExports should be fixable with --fix

### ISSUE THREAD

**Issue by bugeats** [benmosher/eslint-plugin-import#1473]
Oh man I do not want to manually fix all the unused exports in the code base I'm currently looking at. Seems like an easy job for `--fix`.

**Comment by ljharb** [comment]
It seems potentially very destructive to remove unused exports, which might actually be used.

**Comment by bugeats** [comment]
> It seems potentially very destructive to remove unused exports, which might actually be used.

Unused exports are by definition... *unused*. If there's false positives from the AST then we have way bigger problems.

`--fix`'ing a line like:

```javascript
export const foo = 'FOO';
```

to

```javascript
const foo = 'FOO';
```

actually seems rather easy and safe.

**Comment by ljharb** [comment]
In an app, sure - but eslint runs on libraries as often as on apps.

**Comment by bugeats** [comment]
Only libraries without unit tests would suffer from this problem.

**Comment by ljharb** [comment]
@bugeats on the contrary; unit tests would test the individual pieces of a library; there are often legacy entry points that are not tested, but not removed to avoid a breaking change.

Either way, an unsafe autofix is unsafe even if the reason it breaks is because someone is not following coding practices you like.

**Comment by threehams** [comment]
eslint unfortunately doesn't allow for per-rule settings for `fix`: https://github.com/eslint/eslint/issues/7549

While this fixer would be great when run once (i.e. when you first enable it), it would be extremely annoying to trigger on save during development. This is why none of the "unused" rules have fixers - it's not clear whether the fix is to remove the unused thing _or_ to use the thing that's unused.
